dataset: AIP14 dataset

1MLightyears opened the pull request:
<!--
Thank you 🙏 for your contribution to [Ultralytics](https://www.ultralytics.com/) 🚀! Your effort in enhancing our repositories is greatly appreciated. To streamline the process and assist us in integrating your Pull Request (PR) effectively, please follow these steps:

1. Check for Existing Contributions: Before submitting, kindly explore existing PRs to ensure your contribution is unique and complementary.
2. Link Related Issues: If your PR addresses an open issue, please link it in your submission. This helps us better understand the context and impact of your contribution.
3. Elaborate Your Changes: Clearly articulate the purpose of your PR. Whether it's a bug fix or a new feature, a detailed description aids in a smoother integration process.
4. Ultralytics Contributor License Agreement (CLA): To uphold the quality and integrity of our project, we require all contributors to sign the CLA. Please confirm your agreement by commenting below:

    I have read the CLA Document and I sign the CLA

For more detailed guidance and best practices on contributing, refer to our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Your adherence to these guidelines ensures a faster and more effective review process.
-->

Australian Invasive Plants dataset(AIP14)

Dataset: https://github.com/1MLightyears/AIP14

This dataset contains 1,500 images of 14 most common invasive plants in Australia. Each category contains 100 images. All images are sampled from [iNaturalist.org](https://www.inaturalist.org/) with CC-BY-NC license, and are human-labelled for object detection jobs on YOLO family models.

I have read the CLA Document and I sign the CLA

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds a new dataset configuration file for **AIP14** to make it directly usable with Ultralytics training workflows. 📦

### 📊 Key Changes
- ➕ Introduces `ultralytics/cfg/datasets/AIP14.yaml` as a new dataset YAML
- 🗂️ Defines dataset structure (`path`, `train`, `val`) and 15 class names
- ⬇️ Provides a `download` link to fetch a compressed AIP14 dataset release

### 🎯 Purpose & Impact
- 🚀 Enables quick AIP14 dataset training/validation by referencing a built-in dataset config
- 🧩 Standardizes class naming and directory layout for repeatable experiments
- ⚠️ Note: class index `0` is currently an empty string (`""`), which may cause confusing logs/metrics or class-name display issues—worth confirming if intentional

1MLightyears committed:
add AIP14 dataset into cfg/datasets

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

UltralyticsAssistant reviewed (commented):
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, minimal dataset YAML addition overall (path/train/val and class mapping are straightforward). The main issue is `names[0]` being empty, which is likely to cause confusing outputs and potentially downstream issues; please confirm intent and either provide a real class name at index 0 or re-index the classes accordingly.

💬 Posted 1 inline comment

1MLightyears committed:
change spiked_pepper from category 14 to category 0, avoiding placeholder category 0

1MLightyears committed:
resolve conflict in AIP14.yaml

1MLightyears committed:
Merge commit 'a2bf34fa', resolving conflicts

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com/actions

glenn-jocher committed:
Merge branch 'main' into main

1MLightyears commented:
Hi I have updated the PR and fixed the issued mentioned above. Any comments on that? I noticed commit 8f9aaef, the merging to main branch, but I couldn't find the yaml file under `ultralytics/cfg/datasets`.

glenn-jocher commented:
`8f9aaef` is just a sync-merge of `ultralytics/main` into your PR branch (the PR is still open), so you won’t see `ultralytics/cfg/datasets/AIP14.yaml` on the `main` branch until this PR is merged; you can view it in the [Files changed tab](https://github.com/ultralytics/ultralytics/pull/23033/files) (or directly as [AIP14.yaml in this PR](https://github.com/ultralytics/ultralytics/blob/8f9aaef10736a08cb56637ea6072fb0a88ff684c/ultralytics/cfg/datasets/AIP14.yaml)). One quick nit: the `train`/`val` “4 images” comments look like leftover template text—please update/remove them, and ensure the release zip extracts to `datasets/AIP14/` to match `path: AIP14` (otherwise autodownload will fail as described in the [check_det_dataset reference](https://docs.ultralytics.com/reference/data/utils/)); for general PR expectations please also see the [contributing guide](https://docs.ultralytics.com/help/contributing/).

1MLightyears committed:
Merge branch 'ultralytics:main' into main

1MLightyears commented:
> `8f9aaef` is just a sync-merge of `ultralytics/main` into your PR branch (the PR is still open), so you won’t see `ultralytics/cfg/datasets/AIP14.yaml` on the `main` branch until this PR is merged; you can view it in the [Files changed tab](https://github.com/ultralytics/ultralytics/pull/23033/files) (or directly as [AIP14.yaml in this PR](https://github.com/ultralytics/ultralytics/blob/8f9aaef10736a08cb56637ea6072fb0a88ff684c/ultralytics/cfg/datasets/AIP14.yaml)). One quick nit: the `train`/`val` “4 images” comments look like leftover template text—please update/remove them, and ensure the release zip extracts to `datasets/AIP14/` to match `path: AIP14` (otherwise autodownload will fail as described in the [check_det_dataset reference](https://docs.ultralytics.com/reference/data/utils/)); for general PR expectations please also see the [contributing guide](https://docs.ultralytics.com/help/contributing/).

Thank you for your comments! I have updated the data description file `AIP14.yaml`, removing irrelavent comments. I also have double checked the structure, ensuring once the zip is deflated, the folder should look like
```
AIP14.yaml
datasets
└── AIP14
        └── images
                └── train
                └── val
        └── labels
                └── train
                └── val
```
It seems more appropriate. 
Looking forward to see this PR merged.

glenn-jocher commented:
Thanks — that folder layout matches what Ultralytics expects for `path: AIP14`; before we merge, please just ensure the GitHub release asset unzips *directly* into `datasets/AIP14/` (no extra top-level nesting) and that `names` matches the actual 14 classes (to avoid `nc`/`names` mismatches); the expected structure is described in the [Ultralytics datasets guide](https://docs.ultralytics.com/datasets/).

```python
from ultralytics.data.utils import check_det_dataset

d = check_det_dataset("ultralytics/cfg/datasets/AIP14.yaml", autodownload=False)
print(d["path"], d["train"], d["val"], d["nc"])
```

If CI is green after your last push, we’ll take another look and merge when ready (no ETA).
